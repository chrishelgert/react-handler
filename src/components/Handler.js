import React, { PropTypes } from 'react';
import EmptyHandler from './EmptyHandler';
import ErrorHandler from './ErrorHandler';
import LoadingHandler from './LoadingHandler';

/**
 * The Handler handles loading, error and empty state
 * or display the given Component.
 */
const Handler = ({
  LoadingComponent,
  ErrorComponent,
  EmptyComponent,
  Component,
  loadingMessage,
  errorMessage,
  emptyMessage = '',
  emptyPropertyKey,
  componentProps = {},
  loading = true,
}) => (
  <div className="handler">
    <LoadingHandler
      loading={loading}
      message={loadingMessage}
      LoadingComponent={LoadingComponent}
    />

    <ErrorHandler message={errorMessage} ErrorComponent={ErrorComponent} />

    <EmptyHandler
      message={emptyMessage}
      emptyPropertyKey={emptyPropertyKey}
      EmptyComponent={EmptyComponent}
      componentProps={componentProps}
      Component={Component}
    />
  </div>
);

Handler.propTypes = {
  LoadingComponent: PropTypes.func,
  ErrorComponent: PropTypes.func,
  EmptyComponent: PropTypes.func,
  Component: PropTypes.func.isRequired,
  loadingMessage: PropTypes.string,
  errorMessage: PropTypes.string,
  emptyMessage: PropTypes.string,
  emptyPropertyKey: PropTypes.string,
  componentProps: PropTypes.object,
  loading: PropTypes.bool,
};

export default Handler;
