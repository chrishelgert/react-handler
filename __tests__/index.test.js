import Handler, * as SupHandlers from '../src'

describe('react-handler', () => {
  test('contains the Handler-Component', () => {
    expect(Handler).toMatchSnapshot()
  })

  test('contains the the SupHandlerÂ´s', () => {
    expect(SupHandlers).toMatchSnapshot()
  })
})
