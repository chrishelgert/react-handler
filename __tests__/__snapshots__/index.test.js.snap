// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-handler contains the Handler-Component 1`] = `[Function]`;

exports[`react-handler contains the the SupHandlerÂ´s 1`] = `
Object {
  "EmptyHandler": [Function],
  "ErrorEmptyHandler": [Function],
  "ErrorHandler": [Function],
  "LoadingHandler": [Function],
  "default": [Function],
}
`;
